docs: moving to MkDocs

bladeacer opened the pull request:
<!--
  For Work In Progress Pull Requests, please use the Draft PR feature,
  see https://github.blog/2019-02-14-introducing-draft-pull-requests/ for further details.

  For a timely review/response, please avoid force-pushing additional
  commits if your PR already received reviews or comments.

  Before submitting a Pull Request, please ensure you've done the following:
  - 📖 Read the our Code of Conduct: https://github.com/caffeine-addictt/video-manager/blob/main/CODE_OF_CONDUCT.md
  - 👷‍♀️ Create small PRs. In most cases this will be possible.
  - ✅ Provide tests for your changes.
  - 📝 Use descriptive commit messages.
  - 📗 Update any related documentation and include any relevant screenshots.
-->

## What type of PR is this? (check all applicable)

- [ ] Refactor
- [x] Feature
- [ ] Bug Fix
- [ ] Optimization
- [ ] Breaking Change
- [x] Documentation Update



## Description
An Mkdocs site for the documentation. Some links will be broken because not all translations are complete.

It falls back to the English translation of the docs when a missing file is detected. (There are a few)


## Related Tickets & Documents

<!--
For pull requests that relate or close an issue, please include them
below.  We like to follow [Github's guidance on linking issues to pull requests](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue).

For example having the text: "closes #1234" would connect the current pull
request to issue 1234.  And when we merge the pull request, Github will
automatically close the issue.
-->

- Related Issue #54 
- Closes #54 


## QA Instructions, Screenshots, Recordings

_Please replace this line with instructions on how to test your changes, a note
on the devices and browsers this has been tested on, as well as any relevant
images for UI changes._

<img width="2772" height="1628" alt="Screenshot from 2025-09-27 15-50-58" src="https://github.com/user-attachments/assets/699ead4b-54b2-4c42-9478-774bf4806dc4" />

N/A. Just run the setup script activate the Python `.venv` for the site. 

MkDocs itself is highly battle tested being used in many OSS projects.

## Added/updated tests?
_We encourage you to keep the code coverage percentage at 80% and above._

- [ ] Yes
- [x] No, and this is why: Translations themselves are currently incomplete.
- [ ] I need help with writing tests



## [optional] Are there any post deployment tasks we need to perform?

Deployment options are widely documented online. Mentioned in #54 .

- [ ] Update Makefile
- [ ] Update dependabot.yml
- [ ] Deploy site


## Checklist
<!---
Go over all the following points, and put an `x` in all the boxes that apply.
If you're unsure about any of these, don't hesitate to ask. We're here to help!
-->
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.

bladeacer committed:
WIP mkdocs site

bladeacer committed:
Fixed mkdocs site links

caffeine-addictt reviewed (changes requested):
Hey thanks for making this!

Namely:
1. I think that setup.py is unnecessary and that it's better off just moved into the makefile directly. You can try to implement this, otherwise I can at a later time after this is merged.
2. I want to have the python stuff moved into `docs/` if possible, maybe like a `www/docs/` and have them in `www/` like how I have it in waku?
3. Theres some files still using GitHub MD's formatting for INFO or IMPORTANT like below, If i'm not wrong we should have it be in a different syntax for mkdocs to render properly
> [!NOTE]
> test

---

Otherwise, I'm all for merging this in once we can get the above fixed :>

bladeacer commented:
Re:
1. The Makefile should just be for your Golang binary. Idk how I would integrate it with your existing Makefile logic.
2. You can do other than for mkdocs.yml unless you wish for more configuration on your part because mkdocs by default assumes project root and reads the `docs/` directory. Also `www/` is rather late 2000s /lh. As for the Python script it is just a matter of modifying it such that it installs the venv in project root instead of current working directory in which it is called. 
3. Ripgrep yields no admonitions syntax in the current docs. MkDocs' admonitions uses a different syntax, and the current translations are incomplete either way.

Note that I am basing this all off of the main branch.

caffeine-addictt commented:
1. That's fine I can implement that afterwards
2. While yes `www/` is what you'd see from older projects, I'd prefer to have documentation-related stuff OUT of project root in tucked away in its own directory
3.
<img width="397" height="514" alt="image" src="https://github.com/user-attachments/assets/469070c0-cb93-416f-b013-d11cde91667f" />

bladeacer commented:
guess I did not search with caps for the admonitions 😅

will see what I can do with them modifying them to mkdocs' admonitions syntax

bladeacer committed:
Moved docs to www directory

bladeacer commented:
Just updated the docs to use mkdocs admonitions format and moved docs to the www directory as requested.

It is a bit cursed as the widely documented `!!! note` format seems to not be supported for some reason.

Here is an updated guide on how to add admonitions to mkdocs sites: https://discuss.privacyguides.net/t/how-to-use-admonitions-on-the-site/27361

caffeine-addictt reviewed (changes requested):
Could you remove `setup.py`? After that I think we can merge thanks 👍🏻

bladeacer committed:
Removed setup script

bladeacer commented:
I have removed the setup script.

caffeine-addictt commented:
Awesome

caffeine-addictt reviewed (approved):
LGTM
